machinist/state/controlplane.go

```diff
@@ -23,12 +22,16 @@ type MachineController struct {
 func AddMachine(mc *MachineController, m *Machine) error {
 	mc.Lock()
 	defer mc.Unlock()
-	for _, mm := range mc.Machines {
-		if mm.Name == m.Name {
-			return fmt.Errorf("machinist: named machine %s already exists", m.Name)
+	modifiedInPlace := false
+	for i := range mc.Machines {
+		if mc.Machines[i].Name == m.Name {
```

Comment:
Should `Machines` be a `map[string]*Machine` keyed by name instead?

Reply:
So I intentionally left this an array because I didn't want to be stuck on strings as uniqueness. Eventually I want to move to generics or the Equals interface.